diff --git a/drivers/kernelsu/selinux/rules.c b/drivers/kernelsu/selinux/rules.c
index cb1da796..14dd5ab0 100644
--- a/drivers/kernelsu/selinux/rules.c
+++ b/drivers/kernelsu/selinux/rules.c
@@ -173,6 +173,56 @@ void apply_kernelsu_rules()
     ksu_allow(db, "kernel", "apk_data_file", "file", "read");
     ksu_allow(db, "kernel", "apk_data_file", "file", "open");
 
+    // Injecting Tenebrion SELinux rules
+    ksu_allow(db, "kernel", "cgroup", "dir", "getattr");
+    ksu_allow(db, "kernel", "cgroup", "dir", "search");
+    ksu_allow(db, "kernel", "cgroup", "dir", "write");
+    ksu_allow(db, "kernel", "cgroup", "file", "getattr");
+    ksu_allow(db, "kernel", "cgroup", "file", "open");
+    ksu_allow(db, "kernel", "cgroup", "file", "read");
+    ksu_allow(db, "kernel", "cgroup", "file", "write");
+    ksu_allow(db, "kernel", "sysfs", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs", "file", "open");
+    ksu_allow(db, "kernel", "sysfs", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_backlight", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_backlight", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_backlight", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_backlight", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_backlight", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_battery_supply", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_battery_supply", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_battery_supply", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_battery_supply", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_battery_supply", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_drm", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_drm", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_drm", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_drm", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_drm", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_leds", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_leds", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_leds", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_leds", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_leds", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_memory", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_memory", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_memory", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_memory", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_memory", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_memory", "file", "write");
+    ksu_allow(db, "kernel", "sysfs_thermal", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_thermal", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_thermal", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_thermal", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_thermal", "file", "read");
+    ksu_allow(db, "kernel", "sysfs_type", "dir", "getattr");
+    ksu_allow(db, "kernel", "sysfs_type", "dir", "search");
+    ksu_allow(db, "kernel", "sysfs_type", "file", "getattr");
+    ksu_allow(db, "kernel", "sysfs_type", "file", "open");
+    ksu_allow(db, "kernel", "sysfs_type", "file", "read");
+
     rcu_assign_pointer(selinux_state.policy, pol);
     synchronize_rcu();
     ksu_destroy_sepolicy(old_pol);
